scripts/components/CalendarAddEventForm.jsx

```diff
@@ -0,0 +1,167 @@
+var React = require('react');
+var Api = require('../utils/Api.js');
+var MuiContextified = require('./MuiContextified.jsx');
+var mui = require('material-ui');
+var TextField = mui.TextField;
+var RaisedButton = mui.RaisedButton;
+var Dialog = mui.Dialog;
+var Label = require('./Label.jsx');
+var ErrorMessage = require('./ErrorMessage.jsx');
+var Joi = require('joi');
+var JoiError = require('./JoiError.jsx');
+var Radium = require('radium');
+
+var CalendarAddEventForm = React.createClass({
+  propTypes: {
+    eventAdded: React.PropTypes.func,
+    propertyId: React.PropTypes.string,
+  },
+
+  getInitialState() {
+    return {
+      eventDesc: '', // User entered description
+      date: '',
+      time: '',
+      notes: '',
+      validationError: null, // clientside validation error object
+    };
+  },
+
+  onButtonClick() {
+    this.refs.dialog.show();
+  },
+
+  /**
+   * Event handler for capturing in the input field state on each keypress.
+   * @param  {String} field The identifier for the input field.
+   */
+  onChange(field, event) {
+    this.setState({ [field]: event.target.value });
+  },
+
+  /**
+   * Form submission event handler. Sends a request to the server to add the
+   * repair request, and updates the repair requests if successful.
+   */
+  onSubmit(event) {
+
+    // API call to add repair request
+    var propertyId = this.props.propertyId;
+
+    this.setState({
+      validationError: undefined,
+    });
+
+    var validation = this.validate();
+    if (validation.error) {
+      this.setState({ validationError: validation.error });
+      return;
+    }
+
+    Api.addPropertyCalendarEvents({
+      data: {
+        eventDesc: this.state.eventDesc,
+        date: this.state.date,
+        time: this.state.time,
+        notes: this.state.notes,
+        propertyId: propertyId,
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
```

Comment:
Would be best if you notify the user in the case of a server-side error while trying to process the request.